### PR TITLE
Update UIAlertViews and general code cleaning

### PR DESCRIPTION
**Pull request by DerekMelchin**

### CODE REVIEW

File: Episode-5/after/Blapp/app/controllers/PostDetailController.rb [hunk 1]
```diff
@@ -38,4 +38,4 @@ def saveButton
     @saveButton ||= UIBarButtonItem.alloc.initWithBarButtonSystemItem(UIBarButtonSystemItemSave, target:self, action:'save')
   end
 
-end
+end
```
Comment (wndxlori) [inline Episode-5/after/Blapp/app/controllers/PostDetailController.rb]: This is an example of a "white space" change.  Try to filter those out.  They don't really add any value.

File: Episode-5/after/Blapp/app/delegates/PostsControllerAlertViewDelegate.rb [hunk 2]
```diff
@@ -1,18 +1,14 @@
...
-        Post.create({title: alertView.textFieldAtIndex(0).text, content: "", created_at: Time.now, updated_at: Time.now, pseudonym: Pseudonym.first})
-        cdq.save
-        self.tableView.insertRowsAtIndexPaths([NSIndexPath.indexPathForRow(Post.count, inSection:0)], withRowAnimation:UITableViewRowAnimationTop)
-      elsif alertView.title == "Edit Post Title"
-        Post.all[alertView.tag].title = alertView.textFieldAtIndex(0).text
-        Post.all[alertView.tag].updated_at = Time.now
-        cdq.save
-        self.tableView.reloadRowsAtIndexPaths([NSIndexPath.indexPathForRow(alertView.tag + 1, inSection:0)], withRowAnimation:UITableViewRowAnimationFade)
-      end
-    end
```
Comment (wndxlori) [inline Episode-5/after/Blapp/app/delegates/PostsControllerAlertViewDelegate.rb]: This was intended to remove some duplicate code.  See comment below.

File: Episode-5/after/Blapp/app/delegates/PostsControllerTableViewDelegate.rb [hunk 3]
```diff
@@ -4,20 +4,30 @@ def tableView(tableView, didSelectRowAtIndexPath:indexPath)
...
-      alert.show
+      alert = UIAlertController.alertControllerWithTitle('Post Title',
+                                                         message: 'Enter a title for your Blapp post',
+                                                         preferredStyle: UIAlertControllerStyleAlert)
+      cancel_action = UIAlertAction.actionWithTitle('Cancel', style: UIAlertActionStyleCancel, handler: nil)
+      continue_action = UIAlertAction.actionWithTitle('Continue', style: UIAlertActionStyleDefault, handler: lambda { |_| create_new_post(alert.textFields[0].text)})
+      alert.addTextFieldWithConfigurationHandler(lambda {|_|})
+      alert.addAction(cancel_action)
+      alert.addAction(continue_action)
+      presentViewController(alert, animated: true, completion: nil)
```
Comment (wndxlori) [inline Episode-5/after/Blapp/app/delegates/PostsControllerTableViewDelegate.rb]: Chunk 1 (See below)

File: Episode-5/after/Blapp/app/controllers/PostDetailController.rb (hunk 1 above)
Comment (DerekMelchin) [reply Episode-5/after/Blapp/app/controllers/PostDetailController.rb]: Alright

File: Episode-5/after/Blapp/app/delegates/PostsControllerTableViewDelegate.rb [hunk 4]
```diff
@@ -4,20 +4,30 @@ def tableView(tableView, didSelectRowAtIndexPath:indexPath)
...
-    alert.show
+    alert = UIAlertController.alertControllerWithTitle('Edit Post Title',
+                                                       message: 'Enter a title for your Blapp post',
+                                                       preferredStyle: UIAlertControllerStyleAlert)
+    cancel_action = UIAlertAction.actionWithTitle('Cancel', style: UIAlertActionStyleCancel, handler: nil)
+    continue_action = UIAlertAction.actionWithTitle('Continue', style: UIAlertActionStyleDefault, handler: lambda { |_| update_post(indexPath.row - 1, with_title: alert.textFields[0].text)})
+    alert.addTextFieldWithConfigurationHandler(lambda {|text_field| text_field.text = Post.all[indexPath.row - 1].title})
+    alert.addAction(cancel_action)
+    alert.addAction(continue_action)
+    presentViewController(alert, animated: true, completion: nil)
```
Comment (wndxlori) [inline Episode-5/after/Blapp/app/delegates/PostsControllerTableViewDelegate.rb]: Chunk 2.  These are almost duplicate chunks of code.  See if you can figure out how to tease them out into a method that can be called in both places, like the method you deleted above.

**Pull request merged**